File: lib/rest/yampi/mappers/yampi_variation_mapper.dart
import 'package:sertton/core/catalog/dtos/variation_dto.dart';
import 'package:sertton/rest/types/json.dart';

class YampiVariationMapper {
  static VariationDto toDto(Json json) {
    return VariationDto(
      id: json['id'].toString(),
      name: json['name'],
      value: json['value'],
    );
  }

  static List<VariationDto> toDtoList(Json json) {
    final data = (json['data'] as List).cast<Json>();
    return data.map(toDto).toList();
  }
}

File: src/rest/yampi/mappers/yampi-variation-mapper.ts
import type { Variation } from "@/core/catalog/entities"
import type { YampiVariation } from "@/rest/yampi/types"

export const YampiVariationMapper = () => ({
  toDomain(input: YampiVariation): Variation {
    return { id: String(input.id), name: input.name, value: input.value }
  },
})
